---
bool success
string message
string[] names
string[] descriptions